### PR TITLE
Task/form data file

### PR DESCRIPTION
**Pull request by gbudau**
This class encapsulates and parses the body of a POST message containing a file uploaded with an HTML form.

Example of the HTTP request:
```
POST /upload/ HTTP/1.1
(other HTTP headers)
Content-Type: multipart/form-data; boundary=something

--something
Content-Disposition: form-data; name="upload"; filename="file.txt"
Content-Type: text/plain

(The contents of the file)
--something--
```

### CODE REVIEW

File: srcs/app/http/FormFile.cpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
+#include <FormFile.hpp>
+#include <stdexcept>
+#include <StringUtils.hpp>
+
+const char FormFile::kCRLF[] = "\r\n";
+const char FormFile::kWhitespace[] = " \t";
```
[the remark this answers is not in the document]
Comment (gbudau) [reply srcs/app/http/FormFile.cpp:6]: [RFC2046](https://datatracker.ietf.org/doc/html/rfc2046#section-5.1.1) specifies a dictionary for multipart messages, and in this dictionary appears `transport-padding`:
```
transport-padding := *LWSP-char
```
`LWSP-char` appears in [RFC822](https://datatracker.ietf.org/doc/html/rfc822#appendix-D):
```
HTAB        =  <ASCII HT, horizontal-tab>
LWSP-char   =  SPACE / HTAB
```

TLDR: These are the whitespace characters specified in RFC

File: srcs/app/test/http/FormFileTest.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#include <catch2.hpp>
+
+TEST_CASE("ValidFormFile", "[http]") {
+	const std::string raw_request =
+		"POST /foo HTTP/1.1\r\n"
+		"Host: 127.0.0.1:8000\r\n"
+		"Content-Length: 134\r\n"
+		"Content-Type: multipart/form-data; boundary=something\r\n"
+		"\r\n"
+		"--something\r\n"
```
Comment (vdedios) [inline srcs/app/test/http/FormFileTest.cpp:12]: Just a question, in the form boundary you can define any string like here?

File: srcs/app/http/FormFile.cpp [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+/*
+** Example body
+--something\r\n
+Content-Disposition: form-data; name="filename"; filename="foo.txt"\r\n
+Content-Type: text/plain\r\n
+\r\n
+foobar\n\r\n
+--something--\r\n
+*/
+void	FormFile::ParseRequestBody_(const HttpRequest &request) {
```
Comment (vdedios) [inline srcs/app/http/FormFile.cpp:40]: I think this method is only reading the first field of the form?🤔

**Review by vdedios — APPROVED** [review]
LGTM🚀

File: srcs/app/http/FormFile.cpp (hunk 3 above)
Comment (gbudau) [reply srcs/app/http/FormFile.cpp:40]: You are correct. This method is reading the first field of the form, and it requires to be a file. If it has more than one field or is not a file, it throws an error.

File: srcs/app/http/FormFile.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbudau) [reply srcs/app/http/FormFile.cpp:6]: I will make a new PR for that since it's refactoring.
